lib/compat/wordpress-6.0/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * WP_Theme_JSON_Resolver_Gutenberg class
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Class that abstracts the processing of the different data sources
+ * for site-level config and offers an API to work with them.
+ *
+ * This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
+ * This is a low-level API that may need to do breaking changes. Please,
+ * use get_global_settings, get_global_styles, and get_global_stylesheet instead.
+ *
+ * @access private
+ */
+class WP_Theme_JSON_Resolver_Gutenberg extends WP_Theme_JSON_Resolver_5_9 {
```

Comment:
> This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).

If this is the case can it be `final` ? I and @hellofromtonya wonder if we should think of why is this the case though.

Reply:
We still need the ability to modify this behavior by creating child classes in Gutenberg.

Reply:
Then the comment should reflect that? Or something :D

Reply:
For context, this is the full comment:

> * This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
> * This is a low-level API that may need to do breaking changes. Please,
> * use get_global_settings, get_global_styles, and get_global_stylesheet instead.

I think it's the same that we have in [core](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json-resolver.php#L14). I'm happy to iterate on this and use a different wording.